src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs

```diff
@@ -235,6 +239,17 @@ public void Dispose()
                 throw new ProtocolException(
                     string.Format(CultureInfo.CurrentCulture, Strings.Plugin_Exception, plugin.Name, ex.Message));
             }
+            catch (TaskCanceledException ex)
+            {
+                stopwatch.Stop();
+
+                int? exitCode = GetExitCodeOrNull(process);
+
+                plugin.Dispose();
+
+                throw new PluginException(
+                    string.Format(CultureInfo.CurrentCulture, Strings.Plugin_FailedOnCreation, plugin.Name, stopwatch.Elapsed.TotalSeconds, exitCode), ex);
```

Comment:
`TotalSeconds` is a `double`, you might want to round it off in the string resource 

`Plugin '{0}' failed within {1:N2} seconds with exit code {2}.`